bouncer/shared/boost.ts

```diff
@@ -0,0 +1,213 @@
+  });
+  return extrinsicResult;
+}
+
+/// Stops boosting for the given boost pool tier and returns the StoppedBoosting event.
+export async function stopBoosting(
+  asset: Asset,
+  boostTier: number,
+  lpUri = '//LP_BOOST',
+  errorOnFail: boolean = true,
+): Promise<Event | undefined> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  const observeStoppedBoosting = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:StoppedBoosting',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  const extrinsicResult = await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].stopBoosting(
+      shortChainFromAsset(asset).toUpperCase(),
+      boostTier,
+    ),
+    lp,
+    errorOnFail,
+  );
+  if (!extrinsicResult.dispatchError) {
+    console.log('waiting for stop boosting event');
+    return observeStoppedBoosting;
+  }
+  console.log('Already stopped boosting');
+  return undefined;
+}
+
+/// Provides liquidity, then adds the funds to the boost pool of the given tier and returns the BoostFundsAdded event.
+export async function addBoostFunds(
+  asset: Asset,
+  boostTier: number,
+  amount: number,
+  lpUri = '//LP_BOOST',
+): Promise<Event> {
+  await using chainflip = await getChainflipApi();
+  const lp = keyring.createFromUri(lpUri);
+
+  assert(boostTier > 0, 'Boost fee must be greater than 0');
+
+  await provideLiquidity(asset, amount, true, lpUri);
+
+  const observeBoostFundsAdded = observeEvent(
+    chainFromAsset(asset).toLowerCase() + 'IngressEgress:BoostFundsAdded',
+    chainflip,
+    (event) =>
+      event.data.boosterId === lp.address &&
+      event.data.boostPool.asset === asset &&
+      event.data.boostPool.tier === boostTier.toString(),
+  );
+
+  // Add funds to the boost pool
+  console.log(`Adding boost funds of ${amount} ${asset} at ${boostTier}bps`);
+  await lpSubmitExtrinsic(
+    chainflip.tx[ingressEgressPalletForChain(chainFromAsset(asset))].addBoostFunds(
+      shortChainFromAsset(asset).toUpperCase(),
+      amountToFineAmount(amount.toString(), assetDecimals(asset)),
+      boostTier,
+    ),
+    lp,
+  );
+
+  return observeBoostFundsAdded;
+}
+
+/// Adds boost funds to the boost pool and does a swap with boosting enabled, then stops boosting and checks the fees collected are correct.
+async function testBoostingForAsset(asset: Asset, boostFee: number, lpUri: string, amount: number) {
+  await using chainflip = await getChainflipApi();
+  console.log(`Testing boosting for ${asset} at ${boostFee}bps`);
+
+  // Start with a clean slate by stopping boosting before the test
+  const preTestStopBoostingEvent = await stopBoosting(asset, boostFee, lpUri, false);
```

Comment:
Seems weird to "stop boosting" just to get the event to check there aren't any pending boosts. Maybe better to use the new boost rpcs for this? Or just inspect storage?

Reply:
It also cleans up at the end of the test so we don't have and boosting funds left in this tier for other tests that might happen after this.

Reply:
I like to make the tests resistant to failing halfway and getting stuck in a state that needs to be cleaned up manually before the test can be re-run. So i automate the cleanup at the start of the test, instead of just an assert. This also helps a lot when first building the test. The test will only work sequentially for each asset because of the nature of the pools, so i think it's fine.

Reply:
I feel like this was perhaps more useful when you were deveoping this, and not so much anymore (particularly since this doesn't guarantee full reset). I don't really mind keeping it though.